x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -21,8 +26,11 @@ import {
 
 export const executeJobFactory: ExecuteJobFactory<ESQueueWorkerExecuteFn<
   JobDocPayloadPDF
->> = function executeJobFactoryFn(server: ServerFacade) {
-  const generatePdfObservable = generatePdfObservableFactory(server);
+>> = function executeJobFactoryFn(
+  server: ServerFacade,
+  { browserDriverFactory }: { browserDriverFactory?: HeadlessChromiumDriverFactory } = {}
```

Comment:
Feels weird that this is optional. Is there a reason for that?

Reply:
It's because all execute job factory functions are defined by the same type. There is an occurrence of an execute job factory invoked outside of the create_job/execute_job area: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/reporting/server/routes/generate_from_savedobject_immediate.ts#L49 and it wouldn't be appropriate to pass the browser driver factory in there.

More and more, the "immediate" execution flow doesn't feel like the right fit for Reporting area. But I don't see any easy ways out of it.

Reply:
226a34b5a9